### DISCUSSION TITLE
How to specify the initial password through code?

### DISCUSSION THREAD

**Discussion by vikas027** [argoproj/argo-cd#9234] [Q&A] [answered]
I install ArgoCD through code and would like to set the password through code as well.

As of now, I have to patch `admin.passwordMtime` and `admin.password` in the `argocd-secret` secret with a latest date to set the initial password. Is there a way to set the secret via code (date part is tricky).

**Comment by jannfis** [comment, marked as answer]
`admin.passwordMtime` doesn't have to be the time you're patching the resource. It may be any time in the past, basically you could set it UNIX epoch 0. 

The timestamp is set for validation purposes of tokens issued to the admin user - if the token was issued _before_ the time specified in `admin.passwordMtime`, it will be invalid.
